src/group_theory/subgroup.lean

```diff
@@ -619,6 +619,51 @@ end
 lemma closure_singleton_one : closure ({1} : set G) = ⊥ :=
 by simp [eq_bot_iff_forall, mem_closure_singleton]
 
+@[simp, to_additive] lemma closure_inv (S : set G) : closure S⁻¹ = closure S :=
+begin
+  refine le_antisymm ((subgroup.closure_le _).2 _) ((subgroup.closure_le _).2 _),
+  { intros s hs,
+    rw [set_like.mem_coe, ←subgroup.inv_mem_iff],
+    exact subset_closure (mem_inv.1 hs) },
+  { intros s hs,
+    rw [set_like.mem_coe, ←subgroup.inv_mem_iff],
+    exact subset_closure (set.inv_mem_inv.2 hs) },
+end
+
+@[to_additive]
+lemma submonoid.mem_closure_inv (S : set G) (x : G) :
+  x ∈ submonoid.closure S⁻¹ ↔ x⁻¹ ∈ submonoid.closure S :=
+begin
+  suffices : ∀ (S : set G) (x : G), x ∈ submonoid.closure S⁻¹ → x⁻¹ ∈ submonoid.closure S,
+  { refine ⟨this S x, _⟩,
+    have := this S⁻¹ x⁻¹,
+    rw [inv_inv, set.inv_inv] at this,
+    exact this, },
+  intros S x hx,
+  refine submonoid.closure_induction hx (λ x hx, _) _ (λ x y hx hy, _),
+  { exact submonoid.subset_closure (mem_inv.mp hx), },
+  { rw one_inv,
+    exact submonoid.one_mem _ },
+  { rw mul_inv_rev x y,
+    exact submonoid.mul_mem _ hy hx },
+end
```

Comment:
This belongs somewhere in `submonoid.lean` I think

Reply:
It's about `submonoid.closure`, but inside a group, otherwise `⁻¹` doesn't make sense.

Reply:
Sure - but `group` is already available in `submonoid` anyway.

See [this lemma](https://leanprover-community.github.io/mathlib_docs/group_theory/submonoid/membership.html#submonoid.mul_left_mem_add_closure) for instance - it assumes `semiring`, but is still in that file. I think `mem_closure_inv` belongs near that lemma, in `submonoid/membership.lean`.